takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/compile/AbstractCompileMojo.java

```diff
@@ -359,7 +366,7 @@ public Charset getSourceEncoding() {
   public void execute() throws MojoExecutionException, MojoFailureException {
 
     Stopwatch stopwatch = Stopwatch.createStarted();
-
+    context.setFailOnError(failOnError);
```

Comment:
This call requires https://github.com/takari/io.takari.incrementalbuild/commit/84fccedd0f5e88df4d247cc32f5d276f794c0248#diff-db86e3c62a0405d6f9cf318307b4cbde that is present on 0.20.7, hence currently the buid seems broken